### PR TITLE
Abstract Base Class

### PR DESCRIPTION
**Pull request by jhprinz**
This PR should replace PR #302. It uses `abc` from python.

This is ready to review and merge.

### CODE REVIEW

File: openpathsampling/pathmover.py [hunk 1]
```diff
@@ -559,25 +549,32 @@ def _make_backward_trajectory(self, trajectory, shooting_index):
                             trajectory[shooting_index + 1:])
         return trial_trajectory
 
+    # direction is an abstract property to disallow instantiation of the EngineMover unless we use
+    # a concrete subclass that sets this. This is not super elegant but is the way to do it with
+    # abstract classes
```
Comment (dwhswenson) [inline openpathsampling/pathmover.py]: Actually, I think that *is* a pretty elegant way of doing it.

File: openpathsampling/pathmover.py [hunk 2]
```diff
@@ -802,10 +812,9 @@ def __call__(self, sample1, sample2):
 # SUBTRAJECTORY GENERATORS
 ###############################################################################
 
-
-class RandomSubtrajectorySelectMover(SampleMover):
+class SubtrajectorySelectMover(SampleMover):
```
Comment (dwhswenson) [inline openpathsampling/pathmover.py]: I also like the separation of the abstract `SubtrajectorySelect` from `RandomSubtrajSelect`

File: openpathsampling/tests/test_helpers.py [hunk 3]
```diff
@@ -232,4 +239,41 @@ def current_snapshot(self, snapshot):
 
     def generate_next_frame(self):
         self._current_snapshot = None
-        return self.current_snapshot
\ No newline at end of file
+        return self.current_snapshot
+
+
+class IdentityPathMover(paths.PathMover):
```
Comment (dwhswenson) [inline openpathsampling/tests/test_helpers.py]: `IdentityPathMover` might move to the main `pathmover.py`. This is the no-op move I've talked about in #349, so we might use it there.

**Comment by dwhswenson** [comment]
Still waiting for Travis to actually pass it, but this looks pretty good. Two tiny things:

* Do you want to move `IdentityPathMover` into `pathmover.py` as part of this PR? If you don't do it here, I'll probably have to do it later.
* Could you rename `test_pathsimulator.py` to `testpathsimulator.py`? For the most part, our test files don't have the underscore (I think we should also rename `test_toy_dynamics` to `testtoydynamics` to keep it consistent).

File: openpathsampling/tests/test_helpers.py (hunk 3 above)
Comment (jhprinz) [inline openpathsampling/tests/test_helpers.py]: Yes, I had it first in PathMover and then I moved it since it was only part of the test...

**Comment by jhprinz** [comment]
> Do you want to move IdentityPathMover into pathmover.py as part of this PR? If you don't do it here, I'll probably have to do it later.

Sure, no problem. Is `IdentityPathMover` a good name? It was my first idea and seemed to work. It could also be `EmptyPathMover` or `NoPathMover`

> Renaming

Good point. When I named the file I was actually wondering whether to name it with or without the underscore. I would not like to rename all the other files, since it needs to be merged into other branches. So we should do the other renamings later.

**Comment by jhprinz** [comment]
Done...

**Comment by dwhswenson** [comment]
> Is IdentityPathMover a good name?

I think it works well enough. The only thing that is weird about it is that it seems to me like the name suggests an *accepted* no-change. Whether "no change" is accepted or rejected is just semantics, although I'd rather keep it rejected.

I'll merge once tests pass.

**Comment by dwhswenson** [comment]
Genuine test failure: I think you need to remove `paths.` from `IdentityPathMover`'s inheritance.

**Comment by dwhswenson** [comment]
tests pass now! merging.

**Pull request merged**